### ISSUE TITLE
Layout system

### ISSUE THREAD

**Issue by KSemenenko** [jsuarezruiz/AlohaKit.UI#4]
Let's add drawable Layouts like Grid, or Stack or so.
And then we can do like this:

``` xaml
<alohakit:CanvasView> 
   <alohakit:StackLayout>
      <alohakit:RoundRectangle
          WidthRequest="50" HeightRequest="50" 
          CornerRadius="12, 0, 0, 24"
          Fill="Orange" />
          <alohakit:Grid>
            <alohakit:RoundRectangle
                WidthRequest="50" HeightRequest="50" 
                CornerRadius="12, 0, 0, 24"
                Fill="Orange" />
          </alohakit:Grid>
   </alohakit:StackLayout> 
</alohakit:CanvasView>
```
Of course naming is random, but the idea is - to have one canvas with a lot if graphic introls inside.

**Comment by KSemenenko** [comment]
so we no need X and Y now, also we can create complicated laytous. 
@jsuarezruiz  can you guid me plese do you have some like 
**SizeRequest OnMeasure(double widthConstraint, double heightConstraint)** functions?

**Comment by Martinn2** [comment]
We definitely need this, according to me ScrollView is also a layout we must have (besides Stack and Grid).

**Comment by KSemenenko** [comment]
I have code for it, it was implemented some times ago with skiasharp.
But I need some guidance how we can integrate it in better way

**Comment by jsuarezruiz** [comment]
> so we no need X and Y now, also we can create complicated laytous.
> @jsuarezruiz can you guid me plese do you have some like
> **SizeRequest OnMeasure(double widthConstraint, double heightConstraint)** functions?

Definitely, after having gestures and other important Core APIs, Layouts is the next big step forward. I am going to prepare a PR with a basic prototype and we can iterate continue from it. Sounds good?

**Comment by KSemenenko** [comment]
> > so we no need X and Y now, also we can create complicated laytous.
> > @jsuarezruiz can you guid me plese do you have some like
> > **SizeRequest OnMeasure(double widthConstraint, double heightConstraint)** functions?
> 
> Definitely, after having gestures and other important Core APIs, Layouts is the next big step forward. I am going to prepare a PR with a basic prototype and we can iterate continue from it. Sounds good?

Sound good! Waiting for it

**Comment by nextcodelab** [comment]
Maybe Im waiting to have Parent layout like scrollview.. stack,,, gridview, listview... from drawings... I am tired and thinking go to flutter because of lag in .net maui xaml that undergo 50 compilation before it landed on native UI.

**Comment by roughiain** [comment]
Any update?

**Comment by jsuarezruiz** [comment]
Very soon.
![alohakit-layout-grid](https://user-images.githubusercontent.com/6755973/222121777-a9554148-ccf6-40a8-aabe-d960f23e9a62.PNG)

**Comment by nextcodelab** [comment]
> Very soon. ![alohakit-layout-grid](https://user-images.githubusercontent.com/6755973/222121777-a9554148-ccf6-40a8-aabe-d960f23e9a62.PNG)

I like this,,, thanks for this,, I hope every UI Control is breaking down for single task to make it more performant like flutter.
Column => for only column task, 
Row => for only row task
 Expand => to expand available space
Fraction => Divide children base on available space.
Divider => for divider task,
SizeBox => for spacing.
Padding => for padding task
Center => for centering task

**Comment by lukewire129** [comment]
Do you have any plans for an update at the moment?
